### PR TITLE
Tn/minimal violation penalty

### PR DESCRIPTION
**Pull request by Thiago-NovaesB**
- Adds soft constraints for minimal volume and outflow;
- Replace the name of unused variables by "_".

### CODE REVIEW

File: testcases/brasil_4/hydro_dict.jl [hunk 1]
```diff
@@ -22,10 +24,12 @@ function hydrogenerator_dict(;
         "initial_volume" => initial_volume,
         "production_factor" => production_factor,
         "spill_cost" => spill_cost,
+        "minimal_outflow_violation_cost" => minimal_outflow_violation_cost,
+        "minimal_volume_violation_cost" => minimal_volume_violation_cost,
         "downstream_turn" => downstream_turn,
         "downstream_spill" => downstream_spill,
     )
-    if name != nothing
+    if name !== nothing
```
Comment (andrewrosemberg) [inline testcases/brasil_4/hydro_dict.jl]:
```suggestion
    if !isnothing(name)
```
Maybe we can already change to `isnothing`.

**Comment by andrewrosemberg** [comment]
Thank you @Thiago-NovaesB . This looks great.

I am trying to learn how to allow for contributors to run the CI workflow.

**Comment by andrewrosemberg** [comment]
FYI The code will soon move to a new home: [LAMPSPUC](https://github.com/LAMPSPUC/HydroPowerModels.jl).

It is not yet maintained there so I understand if you prefer to merge here. But in case you already want to direct you PRs there, I can review them there too.

**Comment by Thiago-NovaesB** [comment]
Is the version in the LAMPS repository the same as here? I can open PR there, no problem

**Comment by andrewrosemberg** [comment]
> Is the version in the LAMPS repository the same as here? I can open PR there, no problem

Yes and will be the one being updated from now on. The documentation and workflow is still not working there but will be soon.

**Comment by Thiago-NovaesB** [comment]
I deleted the fork I had made, so I could create another fork with the same name

**Pull request closed without merging**